loginserver.js

```diff
@@ -17,20 +17,17 @@ const url = require('url');
 
 const FS = require('./fs');
 
-let TimeoutError = function (message) {
-	Error.captureStackTrace(this, TimeoutError);
-	this.name = "TimeoutError";
-	this.message = message || "";
-};
-TimeoutError.prototype = Object.create(Error.prototype);
-TimeoutError.prototype.constructor = TimeoutError;
-TimeoutError.prototype.toString = function () {
-	if (!this.message) return this.name;
-	return this.name + ": " + this.message;
-};
+/**
+ * A custom error type used when requests to the login server take too long.
+ */
+class TimeoutError extends Error {}
+Object.defineProperty(TimeoutError.prototype, 'name', Object.assign(
```

Comment:
I don't mind spec compliance, but I'm not going to sacrifice readability over spec compliance if the difference is something that's never going to come up in TypeScript-legal code.